articles/rules/redirect.md

```diff
@@ -13,7 +13,7 @@ Rules can also be used to programatically redirect users before an authenticatio
 * Forcing users to change passwords.
 
 ::: panel-danger Caution
-Redirect rules won't work for the [Resource Owner endpoint](/api/authentication/reference#resource-owner) authentication endpoint. You can detect resource owner logins from a rule by checking `context.protocol === 'oauth2-resource-owner'`.
+Redirect rules won't work for the [Resource Owner endpoint](/api/authentication/reference#resource-owner) or the [Refresh Token exchange](/tokens/preview/refresh-token#rules). You can detect resource owner logins from a rule by checking `context.protocol === 'oauth2-password'`. To detect a Refresh Token exchange, check `context.protocol === 'oauth2-refresh-token'`.
 :::
```

Comment:
One thing here is that the OLD Resource Owner endpoint is not going away, so we need to make sure its mention stays in this doc. We need to add oauth/token refresh token and password exchanges. Customers can check oauth2-resource-owner if it's the oauth/ro endpoint, or oauth2-password if oauth/token (password) or oauth2-refresh-token if oauth/token (refresh_token)